src/realm/util/file.hpp

```diff
@@ -885,8 +885,17 @@ class File::AccessError : public std::runtime_error {
     /// no associated file system path, or if the file system path is unknown.
     std::string get_path() const;
 
+    const char* what() const noexcept
+    {
+        m_buffer = std::runtime_error::what();
+        if (m_path.size() > 0)
+            m_buffer += (std::string(" Path: ") + m_path);
+        return m_buffer.c_str();
+    }
+
```

Comment:
@jedelbo, are you aware that this construct makes the exception class not thread-compatible?

Modern C++ paradigm dictates these rules:

A necessary, but insufficient requirement for a type to be *thread-compatible*, is that all its `const` methods are thread-safe.

A class should be considered thread-compatible unless otherwise documented. In general, the vast majority of class should be thread-compatible.

Reply:
Yes, I was aware of that.